### PR TITLE
OCaml 5: resolve remaining mli conflicts

### PR DESCRIPTION
**Pull request by ncik-roberts**
Finish resolving all mli conflicts. I'm not sure if anyone is blocked on these files in particular, but it's nice to just be done resolving mlis so almost everything can be built.

Testing:

I couldn't test `toplevel/topcommon.mli`, `otherlibs/systhreads/thread.mli`, or `ocamltest/ocamltest_config.ml`, as these are built with another dune profile that relies on the compiler being able to build the stdlib. I think the conflict resolutions are simple enough that we should just merge them as-is.

```
# asmcomp/mach.mli
$ dune b --workspace=duneconf/boot.ws _build/default/.ocamloptcomp.objs/byte/mach.cmi

# bytecomp/bytelink.mli
$ dune b --workspace=duneconf/boot.ws _build/default/.ocamlbytecomp.objs/byte/bytelink.cmi

# middle_end/flambda/flambda_utils.mli
$ dune b --workspace=duneconf/boot.ws _build/default/.ocamloptcomp.objs/byte/flambda_utils.cmi
```

### CODE REVIEW

File: middle_end/flambda/flambda_utils.mli [hunk 1]
```diff
@@ -69,13 +69,9 @@ val make_closure_declaration
   -> region:bool
   -> body:Flambda.t
   -> params:Parameter.t list
-<<<<<<< HEAD
   -> return_layout:Lambda.layout
   -> free_variables:Lambda.layout Variable.Map.t
-||||||| merged common ancestors
   -> stub:bool
```
Comment (goldfirere) [inline middle_end/flambda/flambda_utils.mli:74]: Looks like both we and they removed this parameter. Drop it?

File: ocamltest/ocamltest_config.mli [hunk 2]
```diff
@@ -138,7 +128,6 @@ val function_sections : bool
 val instrumented_runtime : bool
 (** Whether the instrumented runtime is available *)
 
-<<<<<<< HEAD
 val naked_pointers : bool
```
Comment (goldfirere) [inline ocamltest/ocamltest_config.mli:131]: This will get removed with the ocaml 5 runtime, I think. In any case, upstream removed it.

**Review by goldfirere — APPROVED** [review]

File: middle_end/flambda/flambda_utils.mli (hunk 1 above)
Comment (ncik-roberts) [reply middle_end/flambda/flambda_utils.mli]: Good catch, thanks. That was a mistake.

File: ocamltest/ocamltest_config.mli (hunk 2 above)
Comment (ncik-roberts) [reply ocamltest/ocamltest_config.mli:135]: I wasn't sure whether we'd want to keep it around until we've gotten around to deleting the flag -- I'll leave this as a `CR ocaml 5 runtime`.

**Pull request merged**